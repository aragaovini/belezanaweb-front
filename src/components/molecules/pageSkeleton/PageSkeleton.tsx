import React from 'react'
import { SkeletonTheme } from 'react-loading-skeleton'
import { BagDetailsSkeleton } from '../bagDetails'
import { ProductPaperSkeleton } from '../productPaper'
import { Container } from './PageSkeleton.style'

export const PageSkeleton = () => (
  <SkeletonTheme baseColor="#d7d7d7" highlightColor="#c9c9c9">
    <Container>
      <ProductPaperSkeleton />
      <BagDetailsSkeleton />
    </Container>
  </SkeletonTheme>
)
