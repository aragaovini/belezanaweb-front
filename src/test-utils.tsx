import React, { ComponentType, ReactElement, ReactNode } from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { defaultTheme } from './global/styles/theme'

interface AllTheProvidersProps {
  children: ReactNode
}

const AllTheProviders = ({ children }: AllTheProvidersProps) => (
  <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
)

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders as ComponentType, ...options })

export * from '@testing-library/react'

export { customRender as render }
