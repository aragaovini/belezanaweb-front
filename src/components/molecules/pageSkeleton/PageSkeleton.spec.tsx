import React from 'react'
import { render } from '../../../test-utils'
import { PageSkeleton } from './PageSkeleton'

const renderComponent = () => render(<PageSkeleton />)

describe('PageSkeleton', () => {
  test('renders correctly', () => {
    const { baseElement } = renderComponent()
    expect(baseElement).toMatchSnapshot()
  })
})
