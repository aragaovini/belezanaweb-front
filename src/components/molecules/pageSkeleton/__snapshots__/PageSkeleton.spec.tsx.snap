// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSkeleton renders correctly 1`] = `
<body>
  <div>
    <div
      class="sc-dIouRR bxyfEO"
    >
      <section
        class="sc-bjUoiL ddtJdZ"
      >
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton"
            style="margin-bottom: 5px; width: 210px; height: 18px; --base-color: #d7d7d7; --highlight-color: #c9c9c9;"
          >
            ‌
          </span>
          <br />
        </span>
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton"
            style="height: 320px; --base-color: #d7d7d7; --highlight-color: #c9c9c9;"
          >
            ‌
          </span>
          <br />
        </span>
      </section>
      <section
        class="sc-jqUVSM iwggig"
      >
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton"
            style="height: 130px; --base-color: #d7d7d7; --highlight-color: #c9c9c9;"
          >
            ‌
          </span>
          <br />
        </span>
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton"
            style="height: 60px; --base-color: #d7d7d7; --highlight-color: #c9c9c9;"
          >
            ‌
          </span>
          <br />
        </span>
      </section>
    </div>
  </div>
</body>
`;
